rfd/0034-desktop-access-windows.md

```diff
@@ -0,0 +1,256 @@
+## Details
+
+### Architecture
+
+Windows Desktop Access is implemented using `windows_desktop_service` that
+translates the Teleport desktop protocol into RDP:
+
+```
++--------+
+| web UI |
++--------+
+    ^
+    | desktop protocol over websocket
+    v
++--------+
+| proxy  |
++--------+
+    ^
+    | desktop protocol over mTLS
+    v
++-------------------------+
+| windows_backend_service |--------------\
++-------------------------+-\            |
+   ^                        |            |
+   | RDP                    | RDP        | LDAP
+   v                        v            |
++---------------------+  +-----------+  +-------------------+
+| windows server 2012 |  | windows 7 |  | domain controller |
++---------------------+  +-----------+  +-------------------+
+```
+
+`windows_desktop_service` can talk to any number of remote Windows RDP hosts.
+It can also talk to `localhost` RDP service, if installed on a Windows machine
+in agent mode (described below).
+
+`windows_desktop_service` also discovers all available Windows hosts from
+Active Directory and registers them in Teleport as `WindowsDesktop` objects.
+
+### Supported versions
+
+Windows Desktop Access supports Windows Server 2012 R2, Windows 7 and newer
+versions. However, it should also work with any server that implements RDP and
+supports Smart Card authentication.
+
+### Protocol
+
+Windows Desktop Access uses Remote Desktop Protocol (RDP). RDP service is built
+into Windows and requires no additional software to be installed on the hosts.
+
+RDP is a very complex protocol, with cruft accumulated over its long history
+and frequent vulnerability discoveries (see [Security
+Concerns](#security-concerns) below). It was not my first choice for this
+implementation. However, it is the only option for supporting concurrent
+sessions and allows easier customer rollout without per-host agents.
+
+#### RDP client
+
+`windows_desktop_service` implements an RDP client. There are no existing Go
+libraries implementing enough of the RDP spec for a basic desktop session.
+
+There are a few options, in order of preference:
+- use [rdp-rs Rust library](https://crates.io/crates/rdp-rs) via FFI from Go,
+  assuming it supports smart cards
+- use [libfreerdp](https://www.freerdp.com/) via CGO, assuming it supports
+  smart cards and concurrent outbound connections
+- implement RDP protocol in Go from scratch; last resort due to the amount of
+  work needed
+
+### Modes
+
+`windows_desktop_service` can run in two modes: gateway and agent.
+
+#### Gateway mode
+
+Gateway mode is for connecting to multiple remote Windows hosts, similar to
+Kubernetes Access. This mode is shown in the [Architecture
+diagram](#architecture) above.
+
+This mode is easy for admins to set up and has minimal resource overhead.
+However, this requires the Windows hosts to expose RDP ports to the
+`windows_desktop_service`, which is more risky (see [Security
+Concerns](#security-concerns) below).
+
+#### Agent mode
```

Comment:
The overall design of agent mode introduces the risk of privilege escalation vulnerabilities since the service has to run as `system`. This is something would need to be throughly tested during the security audit.

Reply:
Agreed :+1: